BasicSample/Assets/Interaction/Scripts/HapticsSupport.cs

```diff
@@ -0,0 +1,29 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.XR;
+
+namespace Microsoft.MixedReality.OpenXR.BasicSample
+{
+    public class HapticsSupport : MonoBehaviour
+    {
+        private readonly List<InputDevice> controllerInputDevices = new List<InputDevice>();
+
+        // Update is called once per frame
+        void Update()
+        {
+            InputDevices.GetDevicesWithCharacteristics(InputDeviceCharacteristics.Controller, controllerInputDevices);
+
+            foreach (InputDevice controller in controllerInputDevices)
+            {
+                if (controller.TryGetFeatureValue(CommonUsages.trigger, out float trigger) && trigger > 0)
+                {
+                    if (controller.TryGetHapticCapabilities(out HapticCapabilities hapticCapabilities) && hapticCapabilities.supportsImpulse)
+                    {
+                        controller.SendHapticImpulse(0, trigger);
```

Comment:
Could you add a comment explaining what this is doing?
Especially like what that `0` means - as someone with no context, this is the part of the code I look at to learn; a comment would be a great help.

Just a guess: is this a `0` second impulse with `trigger` intensity? Otherwise, how would the haptics turn off?

Reply:
Done